### DISCUSSION TITLE
ShellCommand and deprecated 'lazylogfiles' help

### DISCUSSION THREAD

**Discussion by tonyhutter** [buildbot/buildbot#5954] [Q&A]
I'm upgrading from 0.8.x to 3.0.x, and have this:

```python
test_factory = util.BuildFactory()

test_factory.addStep(ShellCommand(
    env={'PATH' : bin_path},
    command=["runurl", bb_url + "bb-dependencies.sh"],
    decodeRC={0 : SUCCESS, 1 : FAILURE, 2 : WARNINGS, 3 : SKIPPED },
    haltOnFailure=True, logEnviron=False,
    lazylogfiles=True,
    description=["installing dependencies"],
    descriptionDone=["installed dependencies"]))
```

Which gives me:
```
2021-03-26 18:38:03+0000 [-]   Invalid argument(s) passed to ShellCommand: lazylogfiles
```
Similarly, `lazylogfiles` isn't working for `Git` either:
```python
test_factory.addStep(Git(repourl=zfs_repo, workdir="build/zfs",
    mode="full", method="clobber", codebase="zfs",
    logEnviron=False, lazylogfiles=True,
    getDescription=True,
    description=["cloning"], descriptionDone=["cloned"],
    doStepIf = do_step_build_zfs,
    hideStepIf=lambda results, s: results==SKIPPED))
```
```
2021-03-26 21:12:48+0000 [-]   <class 'buildbot.steps.source.git.Git'>.__init__ got unexpected keyword argument(s) ['lazylogfiles']
```

I see that `lazylogfiles` has been moved to a `ShellMixin`, but I'm not clear on how to use it.   Any examples on how to use `lazylogfiles` with `Git` and `ShellCommand` would be much appreciated!

**Comment by p12tic** [comment]
`Git` step no longer derives from LoggingBuildStep and no longer supports lazylogfiles. Is this something you require for the Git step?

As for `ShellCommand` you've found a bug, could you file a bug report?

Thanks a lot for raising these issues.

**Comment by tonyhutter** [comment]
Thanks, I've opened a bug: https://github.com/buildbot/buildbot/issues/5957
